internal/app/machined/pkg/runtime/events.go

```diff
@@ -14,15 +18,74 @@ import (
 // Event is what is sent on the wire.
 type Event struct {
 	TypeURL string
+	ID      xid.ID
 	Payload proto.Message
 }
 
 // WatchFunc defines the watcher callback function.
 type WatchFunc func(<-chan Event)
 
+// WatchOptions defines options for the watch call.
+//
+// Only one of TailEvents, TailID or TailDuration should be non-zero.
+type WatchOptions struct {
+	// Return that many past events.
+	//
+	// If TailEvents is negative, return all the events available.
+	TailEvents int
```

Comment:
any suggestions on better names?

Reply:
I always like super descriptive names for public facing things. This works. Just throwing out more ideas since you asked:

```
NumberOfHistoricalEvents
EventsCount
EventsInHistory
```